app/products/calls/screens/call_screen/call_screen.tsx

```diff
@@ -0,0 +1,545 @@
+    },
+}));
+
+const CallScreen = ({currentCall, participantsDict, teammateNameDisplay}: Props) => {
+    const intl = useIntl();
+    const theme = useTheme();
+    const insets = useSafeAreaInsets();
+    const {width, height} = useWindowDimensions();
+    const isLandscape = width > height;
+    const [showControlsInLandscape, setShowControlsInLandscape] = useState(false);
+    const myParticipant = currentCall?.participants[currentCall.myUserId];
+    const style = getStyleSheet(theme);
+    const showControls = !isLandscape || showControlsInLandscape;
+
+    useEffect(() => {
+        mergeNavigationOptions('Call', {
+            layout: {
+                componentBackgroundColor: 'black',
+            },
+            topBar: {
+                visible: false,
+            },
+        });
+    }, []);
+
+    const [speaker, setSpeaker] = useState('');
+    useEffect(() => {
+        const handleVoiceOn = (data: VoiceEventData) => {
+            if (data.channelId === currentCall?.channelId) {
+                setSpeaker(data.userId);
+            }
+        };
+        const handleVoiceOff = (data: VoiceEventData) => {
+            if (data.channelId === currentCall?.channelId && ((speaker === data.userId) || !speaker)) {
+                setSpeaker('');
+            }
+        };
+
+        const onVoiceOn = DeviceEventEmitter.addListener(WebsocketEvents.CALLS_USER_VOICE_ON, handleVoiceOn);
+        const onVoiceOff = DeviceEventEmitter.addListener(WebsocketEvents.CALLS_USER_VOICE_OFF, handleVoiceOff);
+        return () => {
+            onVoiceOn.remove();
+            onVoiceOff.remove();
+        };
+    }, []);
+
+    const leaveCallHandler = useCallback(() => {
+        popTopScreen();
+        leaveCall();
+    }, []);
+
+    const muteUnmuteHandler = useCallback(() => {
+        if (myParticipant?.muted) {
+            unmuteMyself();
+        } else {
+            muteMyself();
+        }
+    }, [myParticipant?.muted]);
+
+    const toggleRaiseHand = useCallback(() => {
+        const raisedHand = myParticipant?.raisedHand || 0;
+        if (raisedHand > 0) {
+            unraiseHand();
+        } else {
+            raiseHand();
+        }
+    }, [myParticipant?.raisedHand]);
+
+    const toggleControlsInLandscape = useCallback(() => {
+        setShowControlsInLandscape(!showControlsInLandscape);
+    }, [showControlsInLandscape]);
+
+    const switchToThread = useCallback(async () => {
+        Keyboard.dismiss();
+        await dismissBottomSheet();
+        if (!currentCall) {
+            return;
+        }
+
+        const activeUrl = await DatabaseManager.getActiveServerUrl();
+        if (activeUrl === currentCall.serverUrl) {
+            goToScreen(Screens.THREAD, '', {rootId: currentCall.threadId});
+            return;
+        }
+
+        // TODO: this is a temporary solution until we have a proper cross-team thread view.
+        //  https://mattermost.atlassian.net/browse/MM-45752
+        popTopScreen();
```

Comment:
if you are switching servers here, does it make sense to `popToRoot` instead of just popping this screen? the reason behind it is that I believe this screens is pushed on top of the channel screen, switching servers and then pushing the thread screen is fine, but tapping the back button will take you back to the channel screen which is the last channel of the server you switched to and that can be absolutely misleading